### PR TITLE
add OCM prefix for clusterrole and clusterrolebinding

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Referrence: https://github.com/stolostron/backlog/issues/26271

### CODE REVIEW

File: operator/config/rbac/auth_proxy_client_clusterrole.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: rbac.authorization.k8s.io/v1
 kind: ClusterRole
 metadata:
-  name: metrics-reader
+  name: open-cluster-management:metrics-reader
```
Comment (clyang82) [inline operator/config/rbac/auth_proxy_client_clusterrole.yaml:4]: we do not use this role. @morvencao right?

Comment (morvencao) [reply operator/config/rbac/auth_proxy_client_clusterrole.yaml:4]: correct, we will never use this role, so no need to update it. @yanmxa

**Review by morvencao — CHANGES_REQUESTED** [review]
@yanmxa I think we don't add prefix to the role/clusterrole/rolebing/clusterrolebinding to the operator, because they will be installed with OLM, the csv will help to generate the names. We should focus on role/clusterrole/rolebing/clusterrolebinding for manager and agent? right? @clyang82

File: operator/config/rbac/auth_proxy_client_clusterrole.yaml (hunk 1 above)
Comment (yanmxa) [reply operator/config/rbac/auth_proxy_client_clusterrole.yaml:4]: Yes, I haven't seen this `clusterrole` is bound to any object. And I will remove it

**Review by morvencao — COMMENTED** [review]
Also I think you missed role and rolebinding for agent in hosted mode.

**Comment by yanmxa** [comment]
> Also I think you missed role and rolebinding for agent in hosted mode.

Maybe we just need to add the `open-cluster-management` to the `clusterrole` and `clusterrolebinding`.  Cause role and rolbinding is namespaced. We don't need a prefix to indicate where it's from. 

Am I right? @clyang82 @morvencao

**Comment by yanmxa** [comment]
> @yanmxa I think we don't add prefix to the role/clusterrole/rolebing/clusterrolebinding to the operator, because they will be installed with OLM, the csv will help to generate the names. We should focus on role/clusterrole/rolebing/clusterrolebinding for manager and agent? right? @clyang82

Agree!

**Comment by morvencao** [comment]
OK @yanmxa make sense to me! thanks.

File: operator/config/rbac/auth_proxy_client_clusterrole.yaml [hunk 2]
```diff
@@ -1,9 +0,0 @@
-apiVersion: rbac.authorization.k8s.io/v1
-kind: ClusterRole
-metadata:
-  name: metrics-reader
-rules:
-- nonResourceURLs:
-  - "/metrics"
-  verbs:
-  - get
```
Comment (morvencao) [inline operator/config/rbac/auth_proxy_client_clusterrole.yaml:9]: Let's keep it for future usage.

Comment (yanmxa) [reply operator/config/rbac/auth_proxy_client_clusterrole.yaml]: Okay. I rollback the role.

**Review by morvencao — APPROVED** [review]
/lgtm

**Pull request merged**